src/mpi_utils/Entangler.jl

```diff
@@ -105,7 +105,7 @@ mpi_active() =
         Comm_size(COMM_WORLD) > 1
     end
 
-init_mpi() = Init() #threadlevel = :funneled)
+init_mpi() = Init(threadlevel = :funneled)
```

Comment:
As troubleshooting, you should try `MPI_THREAD_MULTIPLE`. The default is thread-single, which is nearly the same as thread-funneled. But if you have race condition into MPI, then you need thread-multiple.

Reply:
Thank you for the suggestion! Unfortunately, the segfault still arise with `threadlevel = :multiple`. On the positive side, I am now able to reproduce the problem locally.